src/reporters/block_based_reporter.ts

```diff
@@ -217,6 +217,13 @@ export class BlockBasedReporter extends BaseReporter {
   async maybeReport(blockNumber: number) {
     const price = await this.priceToReport()
     this.config.metricCollector?.potentialReport(this.config.currencyPair, price)
+    // If it's a heartbeat block.
+    if (this.isHeartbeatCycle(blockNumber)) {
```

Comment:
Does this work if the oracle is not whitelisted during the startup?

I think it'll simply fail here: https://github.com/celo-org/celo-oracle/blob/main/src/reporters/base.ts#L390
I guess we could also fix this in a different PR.

Reply:
I think it'd fail during start-up, as `setOracleInfo` throws if it cannot find the oracle's address in the list of approved oracles.

Reply:
I was thinking this should also support the case case when some oracle client is turned on before getting whitelisted, so it'd have to wait until the governance proposal passes and then start reporting. So that no human coordination has to be arranged.